libs/clients/license-client/src/lib/clients/driving-license-client/services/drivingLicenseClient.service.ts

```diff
@@ -0,0 +1,349 @@
+  constructor(
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
+    private drivingApi: DrivingLicenseApi,
+    private smartApi: SmartSolutionsApi,
+  ) {}
+
+  private checkLicenseValidity(
+    license: DriversLicense,
+  ): LicensePkPassAvailability {
+    if (!license || license.photo === undefined) {
+      return LicensePkPassAvailability.Unknown
+    }
+
+    if (!license.photo.image) {
+      return LicensePkPassAvailability.NotAvailable
+    }
+
+    return LicensePkPassAvailability.Available
+  }
+
+  licenseIsValidForPkPass(payload: unknown): LicensePkPassAvailability {
+    return this.checkLicenseValidity(payload as DriversLicense)
+  }
+
+  private fetchCategories = () => this.drivingApi.getRemarksCodeTable()
+
+  private async fetchLicense(
+    user: User,
+  ): Promise<Result<DriversLicense | null>> {
+    try {
+      const licenseInfo = await this.drivingApi.getCurrentLicenseV5({
+        nationalId: user.nationalId,
+        token: user.authorization.replace(/^bearer /i, ''),
+      })
+      return { ok: true, data: licenseInfo }
+    } catch (e) {
+      let error
+      if (e instanceof FetchError) {
+        //404 - no license for user, still ok!
+        error = {
+          code: 13,
+          message: 'Service failure',
+          data: JSON.stringify(e.body),
+        }
+        this.logger.warn('Expected 200 status', {
+          status: e.status,
+          statusText: e.statusText,
+          category: LOG_CATEGORY,
+        })
+      } else {
+        const unknownError = e as Error
+        error = {
+          code: 99,
+          message: 'Unknown error',
+          data: JSON.stringify(unknownError),
+        }
+        this.logger.warn('Unable to query data', {
+          status: e.status,
+          statusText: e.statusText,
+          category: LOG_CATEGORY,
+        })
+      }
+
+      return {
+        ok: false,
+        error,
+      }
+    }
+  }
+
+  private async createPkPassPayload(
+    license: DriversLicense,
+    remarks?: Array<RemarkCode>,
+  ): Promise<PassDataInput | null> {
+    const inputValues = createPkPassDataInput(license, remarks)
+
+    if (!inputValues) {
+      this.logger.warn('PkPassDataInput creation failed', {
+        category: LOG_CATEGORY,
+      })
+      return null
+    }
+
+    //slice out headers from base64 image string
+    const image = license.photo?.image
+
+    if (!inputValues) return null
+    //Fetch template from api?
+    const payload: PassDataInput = {
+      inputFieldValues: inputValues,
+      expirationDate: license.dateValidTo?.toISOString(),
+      thumbnail: image
+        ? {
+            imageBase64String: image.substring(image.indexOf(',') + 1).trim(),
+          }
+        : null,
+    }
+
+    return payload
+  }
+
+  async getLicense(user: User): Promise<Result<DriversLicense | null>> {
+    const licenseData = await this.fetchLicense(user)
+    if (!licenseData.ok) {
+      this.logger.info(`Drivers license data fetch failed`)
```

Comment:
```suggestion
      this.logger.info(`Drivers license data fetch failed`, { category: LOG_CATEGORY })
```
any more info to log here?